Merge 2.12 to 2.13 july 12 [ci: last-only]

lrytz opened the pull request:
```
$> git log HEAD --oneline -n 1 | cat
80fd88d177 Merge pull request #8179 from exoego/update-jquery

$> git log upstream/2.12.x --oneline -n 1 | cat
47667437e5 unset JAVA_HOME (#8216)

$> export MB=$(git merge-base HEAD upstream/2.12.x)

$> git log --graph --oneline $MB..upstream/2.12.x | cat
*   47667437e5 unset JAVA_HOME (#8216)
|\
| * b5e76cc6ba unset JAVA_HOME
* |   1b8515f938 Backport: list all lambda methods in `$deserializeLambda$` (#8214)
|\ \
| |/
|/|
| * 3b3eaf4805 [backport] Fix LambdaMetaFactoryCall for java-generated LMF
| * 54e6b7f566 [backport] Inline: list all lambda methods in `$deserializeLambda$`
|/
*   bd6d22021e Fix scalac compatibility with JDK 13 (#8210)
|\
| * 0aec26384f Fix scalac compatibility with JDK 13
* |   fa704e5845 Use AdoptOpenJDK JDK 8 for testing [ci: last-only] (#7931)
|\ \
| * | 8f7fdc4ee9 Use AdoptOpenJDK JDK 8 for testing
* | |   d43dc78130 Merge pull request #8167 from SethTisue/whitesource-2.12
|\ \ \
| |/ /
|/| |
| * | df4df5f4cc [nomerge] add sbt-whitesource
| |/
* |   1be284c4f1 Merge pull request #8204 from xuwei-k/issue-11598
|\ \
| * | 347127707e optimize immutable.TreeSet#{max, min}
* | |   1314b4566a Merge pull request #8175 from retronym/topic/write-sig-file
|\ \ \
| |/ /
|/| |
| * | 3ee75e0995 Add a compiler option to write .sig files to disk
|  /
* |   627a110ba2 Merge pull request #8193 from tgodzik/fix-issue-with-comments
|\ \
| |/
|/|
| * 2d3a47723d Fix the issue when completions would break in case comments between qualifier and the point at which we do completion.
|/
*   a995c4dfdc Merge pull request #8189 from hrhino/topic/basic
|\
| * 56f60d18f1 [nomerge] set every sourcesInBase := false
|/
* ddcbd881a2 Merge pull request #8187 from takayahilton/remove-unused-private-collectElems
* fe17b84e75 remove unused private method.

$> git merge ddcbd881a2

$> git merge a995c4dfdc -s ours

$> git merge 1be284c4f1

$> git merge d43dc78130 -s ours

$> git merge bd6d22021e

$> git merge 1b8515f938 -s ours

$> git merge 47667437e5
```

diesalbla committed:
Fixes Issue 493, to avoid unnecessary allocation.

This commit avoids unnecessary memory use noticed
in Issue 493 of https://github.com/scala/scala-dev.

The line: `refinementParents :::= currentBaseClass.parentSymbols`
in each iteration pre-prends `currentBaseClass.parentSymbols` to the
previous `refinementParens`. Thus, at the end of the loop,
the length of `refinementParens` is the sum of the `parentSymbols`
lists obtained from each symbol in `initBaseClasses` which is a
refinement classes. That creates as many cons (`::`) objects.
Moreover, since `parentSymbols` is not a `val` but a `def`, it creates
a list of length `m`, copies it (to prepend it) and the throws it away.

To prevent these allocations, we replace the flattened
`refinedParents` list by a `refinedClasses` list, which
stores the classes whose parents we have not yet looked into.
We just use the `exists` methods of the List class to look for a
refinedClass with at least one parent is the currentBaseClass.

(cherry picked from commit 0db7dd4e251ec7be75f6f8178977faae9c8274e9)

lrytz committed:
[backport] Inline: list all lambda methods in `$deserializeLambda$`

When inlining an IndyLambda from the classpath, ensure that the lambda
impl method is listed in the class's `$deserializeLambda$`.

Backport of 88960039cb.

lrytz committed:
[backport] Fix LambdaMetaFactoryCall for java-generated LMF

Fix issues with Java LMF where the impl method is non-static, or a
constructor.

Test inlining of Java methods that have an LMF.

Backport of 4903f691b0.

joroKr21 committed:
[backport] Performance tweak - avoid mapList in isSubArgs

(cherry picked from commit 401154823a3ebf6ecb86226955aa88c59e74d0a6)

eed3si9n committed:
Use AdoptOpenJDK JDK 8 for testing

Ref scala/scala-dev#587

retronym committed:
Add a compiler option to write .sig files to disk

Refactor `ClassfileWriters` to share code with this new output
type.

Take advantage of this in PipelineMain.

```
$  tail sandbox/{test,client}.scala
==> sandbox/test.scala <==
package p1.p2

case class Test()

==> sandbox/client.scala <==
package p3

class Client {
  new p1.p2.Test()
}

$  qscalac -Youtline -Ypickle-java -Ypickle-write /tmp/out -Ystop-after:pickler -d /tmp/out sandbox/test.scala

$  find /tmp/out
/tmp/out
/tmp/out/p1
/tmp/out/p1/p2
/tmp/out/p1/p2/Test.sig

$  qscalac -cp /tmp/out -d /tmp/out sandbox/client.scala

$  find /tmp/out
/tmp/out
/tmp/out/p3
/tmp/out/p3/Client.class
/tmp/out/p1
/tmp/out/p1/p2
/tmp/out/p1/p2/Test.sig
```

takayahilton committed:
remove unused private method.

lrytz committed:
Merge pull request #8187 from takayahilton/remove-unused-private-collectElems

remove unused private method.

hrhino committed:
[nomerge] set every sourcesInBase := false

This would have saved me a headache, although it would also have been
saved by me being minimally hygienic and not leaving `.scala` files lying
around in my git repos.

lrytz committed:
Merge pull request #8189 from hrhino/topic/basic

[nomerge] set every sourcesInBase := false

tgodzik committed:
Fix the issue when completions would break in case comments between qualifier and the point at which we do completion.

retronym committed:
Merge pull request #8193 from tgodzik/fix-issue-with-comments

Fix presentation compiler having wrong name in CompletionResult when comments between qualifier and pos.

xuwei-k committed:
optimize immutable.TreeSet#{max, min}

retronym committed:
Merge pull request #8175 from retronym/topic/write-sig-file

Add a compiler option to write .sig files to disk

lrytz committed:
Merge pull request #8204 from xuwei-k/issue-11598

optimize immutable.TreeSet#{max, min}

SethTisue committed:
[nomerge] add sbt-whitesource

SethTisue committed:
Merge pull request #8167 from SethTisue/whitesource-2.12

[nomerge] add sbt-whitesource

retronym committed:
Fix scalac compatibility with JDK 13

This change in the JDK:

https://github.com/openjdk/jdk/commit/ac6c642cf4fe243d88c2b762502860fdd41676f4#diff-e9881878ce74700a8063f67f65ec0657

Led us to the "cannot be represented as URI" exception.

This commit just hard-codes jrt:/ as the URI

lrytz committed:
Use AdoptOpenJDK JDK 8 for testing [ci: last-only] (#7931)

Use AdoptOpenJDK JDK 8 for testing [ci: last-only]

lrytz committed:
Fix scalac compatibility with JDK 13 (#8210)

Fix scalac compatibility with JDK 13

eed3si9n committed:
unset JAVA_HOME

lrytz committed:
Backport: list all lambda methods in `$deserializeLambda$` (#8214)

Backport: list all lambda methods in `$deserializeLambda$`

lrytz committed:
unset JAVA_HOME (#8216)

unset JAVA_HOME

lrytz committed:
Merge commit 'ddcbd881a2' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit 'a995c4dfdc' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit '1be284c4f1' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit 'd43dc78130' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit 'bd6d22021e' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit '1b8515f938' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit '47667437e5' into merge-2.12-to-2.13-july-12

eed3si9n commented:
```
[error] /home/travis/build/scala/scala/test/junit/scala/tools/nsc/interpreter/CompletionTest.scala:247:25: not found: type PresentationCompilerCompleter
[error]     val completer = new PresentationCompilerCompleter(intp)
[error]                         ^
[warn] there were 22 deprecation warnings (since 2.13.0); re-run with -deprecation for details
[warn] there were three feature warnings; re-run with -feature for details
[warn] two warnings found
[warn] three warnings found
[error] one error found
[error] (junit / Test / compileIncremental) Compilation failed
```

som-snytt commented:
Why don't we see
```
[warn] there were 22 deprecation warnings (since July 12)
```
I'm having a problem where I let the daughter watch Jane the Virgin, but suddenly the last five minutes won't play with sound.

I only care about a deprecation warning if it lets me figure out how to watch the last five minutes of Jane the Virgin with sound enabled.

retronym committed:
Tweak URI creation in RootPath for better windows support

The `getPath` I used was incomplete, it doesn't include the server
name on windows UNC paths (\\servername\a\b)

I could have tried the `toString` or `toAsciiString`, but after
reading about pitfalls in:

   https://stackoverflow.com/questions/9873845/java-7-zip-file-system-provider-doesnt-seem-to-accept-spaces-in-uri

I opted instead to just find the relevent FileSystemProvider
and use the Path itself.

retronym committed:
Reduce the overhead of active analyzer plugins

lrytz committed:
Merge pull request #8223 from retronym/backport/findMemberAlloc

[backport] Fixes Issue 493, to avoid unnecessary allocation.

lrytz committed:
Merge pull request #8222 from retronym/backport/subArgs

[backport] Performance tweak - avoid mapList in isSubArgs

lrytz committed:
Tweak URI creation in RootPath for better windows support (#8224)

Tweak URI creation in RootPath for better windows support

lrytz committed:
Reduce overhead of active analyzer plugins in hot paths (#8227)

Reduce overhead of active analyzer plugins in hot paths

lrytz committed:
Merge commit '9d17b6ad04' into merge-2.12-to-2.13-july-12

lrytz committed:
Merge commit 'eb9bfc4bc5' into merge-2.12-to-2.13-july-12

lrytz commented:
Included the latest changes that went into 2.12.x

```
$> git log upstream/2.12.x --oneline -n 1 | cat
eb9bfc4bc5 Reduce overhead of active analyzer plugins in hot paths (#8227)
$> export MB=$(git merge-base HEAD upstream/2.12.x)
$> git log --graph --oneline $MB..upstream/2.12.x | cat
*   eb9bfc4bc5 Reduce overhead of active analyzer plugins in hot paths (#8227)
|\
| * b6ce0ce2a6 Reduce the overhead of active analyzer plugins
*   0d9abe7888 Tweak URI creation in RootPath for better windows support (#8224)
|\
| * 6f4f12d38e Tweak URI creation in RootPath for better windows support
*   9d17b6ad04 Merge pull request #8222 from retronym/backport/subArgs
|\
| * f8215506c9 [backport] Performance tweak - avoid mapList in isSubArgs
* 7306ee5130 Merge pull request #8223 from retronym/backport/findMemberAlloc
* 8d537d703d Fixes Issue 493, to avoid unnecessary allocation.

$> git merge 9d17b6ad04 -s ours


$> git merge eb9bfc4bc5
```
